src/Core/src/Platform/Windows/MauiWebView.cs

```diff
@@ -55,6 +67,14 @@ public async void LoadHtml(string? html, string? baseUrl)
 
 				await EnsureCoreWebView2Async();
 
+				if (mapBaseDirectory)
+				{
+					CoreWebView2.SetVirtualHostNameToFolderMapping(
+						LocalHostName,
+						ApplicationPath,
+						Web.WebView2.Core.CoreWebView2HostResourceAccessKind.Allow);
```

Comment:
I think there is a potential security problem here.

Consider this scenario:

1. App uses MAUI WebView and calls `LoadHtml("some html blah blah", baseUrl: null);`
2. This causes MAUI to load the specified HTML, and enable local file mapping for the app, thus enabling the WebView to load arbitrary files from within the app contents. This part is pretty much fine, because it's generally assumed that if you're loading HTML yourself, you know what's in it, and that it's considered sanitized and safe
3. Later, the app calls some `Navigate` API on the WebView, causing the WebView to go to some other URL (say, `https://malware.example.com`
4. But because the mapping is still enabled, I believe this could cause the malicious site to load arbitrary content from within the app. I'm not sure what access the malicious site would have to the contents of what it loaded, but it could potentially cause unintended things to be invoked.

So, I think this change needs to be aware of navigation events and that the access/mapping be changed when the WebView navigates away from one of the "safe" locations.

Reply:
Fair point. I'll change it to hook into Navigation events, check target URL for the local hostname OnNavigating, and call CoreWebView2.ClearVirtualHostNameToFolderMapping(..) if attempting to navigate to any other URL. I guess I will need to re-enable that mapping on any attempt to navigate _back_ to the local host URL, too.

Reply:
@breenbob yeah I think some logic like that would fix this. Thanks!

Reply:
OK, have added this and tested. Looks like a) it works and b) it was needed.

To test I created a sample web page that I hosted somewhere, with the following body:
```html
<body>
    <h3>Test Page</h3>
    <p>I am a remote page.</p>
	<iframe src="https://appdir/simple.html" height="200" width="300" title="App Dir"></iframe>
    <br />
    <a href="https://appdir/simple.html">Link back</a>
</body>
```

I added a link to this URL to my simple.html test page.

With the code changes applied, my test page loads fine, as before. I click the link to the external site, that loads fine, but the iframe pointing to local appdir simple html file fails to load:
![image](https://user-images.githubusercontent.com/6000851/174671921-7a5d2efe-ea5a-4226-8d16-bded474df68d.png)

Good so far. I click the link to take me back to https://appdir/simple.html (i.e. so I am navigating forwards, not using back button) and it works fine.

I am conscious that there are potentially multiple calls to Map the virtual hostname now, as it is done in LoadUrl/Html and in NavigationStarted, but it doesn't seem to cause any issue, and unmapping once still unmaps. I had tried to track its status with a boolean flag but got a bit unwieldy, and WebView2 has no method to check what hosts are mapped.

Only peculiar thing I noticed, as you can see in this gif, is that when navigating _back_ both the NavigationFailed and NavigationSucceeded events fire in the `WebViewHandler.Windows.cs`. Failed first, with host name unknown, then succeeded. Actual navigation works fine.
![devenv_yMvWWWanve](https://user-images.githubusercontent.com/6000851/174673094-af0772e1-1fe8-46c8-8718-ff82e2b8a487.gif)

Out of curisoity, I tried loading the external page with the call to ClearVirtualHostNameToFolderMapping in NavigationStarted commented out, just to see, and the iframe is now able to load the appdir simple html page:
![image](https://user-images.githubusercontent.com/6000851/174672282-aabe28f0-83d0-4666-8f98-d8aa02dda7e4.png)

So clearing the virtual host mapping is 100% necessary. Hopefully good to go now?